### PR TITLE
Clean up radsan test files

### PR DESCRIPTION
**Pull request by davidtrevelyan**
Addresses https://github.com/realtime-sanitizer/radsan/issues/9

### CODE REVIEW

File: compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp [hunk 1]
```diff
@@ -134,18 +134,35 @@ TEST(TestRadsanInterceptors, openDiesWhenRealtime) {
...
+
+  /*
+    The creat function doesn't seem to work on an ubuntu image when the path is
+    in a Docker shared volume. For now, to keep testing convenient with a local
+    Docker container, we just put it somewhere that's not in the shared volume
+    (/tmp). This is volatile and will be cleaned up as soon as the container is
+    stopped. */
+
+#if SANITIZER_LINUX
+  auto const filepath = "/tmp/file.txt";
```
Comment (cjappl) [inline compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp]: If we are throwing something in /tmp/, want to make this file something super specific that can't possibly overwrite something the user may have there?

aka

`/tmp/radsan_test_file.txt`

If it isn't hard, you could also add some random UUID or something to make 100% sure this isn't destructive to the user

**Review by cjappl — COMMENTED** [review]
Would like to see at least a slight change to make the file globally on linux more unique, after that 100% approved.

File: compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp (hunk 1 above)
Comment (davidtrevelyan) [reply compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp]: Yeah, good point! Really, there's too much duplication of "file.txt" everywhere in the tests anyway (we were lazy!), so I'll refactor it to be a function that returns "/tmp/radsan_test_file.txt" on Linux and "./radsan_test_file.txt" on macOS.

**Comment by davidtrevelyan** [comment]
Good to go with this @cjappl ?

File: compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp [hunk 2]
```diff
@@ -33,6 +33,21 @@ using namespace std::chrono_literals;
...
 void *fake_thread_entry_point(void *) { return nullptr; }
+
+/*
+  The creat function doesn't seem to work on an ubuntu Docker image when the
+  path is in a shared volume of the host. For now, to keep testing convenient
+  with a local Docker container, we just put it somewhere that's not in the
+  shared volume (/tmp). This is volatile and will be cleaned up as soon as the
+  container is stopped.
+*/
+constexpr auto temporary_file_path() -> const char * {
```
Comment (cjappl) [inline compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp]: Any reason for the trailing return type? Does that match the rest of llvm's style? (Honest question, I don't know)

**Review by cjappl — APPROVED** [review]
Looks good to me. Make sure the trailing return type matches existing LLVM styling guidelines (or there is another good reason for doing it)

Good stuff!! 🚀

File: compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp (hunk 2 above)
Comment (davidtrevelyan) [reply compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp]: Ah yeah - good question. I did initially play with `constexpr const char * temporary_file_path() {`, but looked at it and thought the trailing return type was easier to understand! We've got a bit of a mixture of styles in here anyway, which isn't great (e.g. snake case `fake_thread_entry_point()` and `temporary_file_path()` vs. camel case `expectRealtimeDeath()` and `expectNonrealtimeSurvival()`). There doesn't seem to be really tight style control in the compiler-rt project (the Google-developed sanitizers have Google style, for example, which is a little different to LLVM style). Not sure if we need to worry about it yet!

**Comment by cjappl** [comment]
As discussed on the call, trailing return type is up to you @davidtrevelyan , submit at will. I will file a trello to go through and run clang-tidy and ensure our code is up to LLVM standards later in the process

File: compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp (hunk 2 above)
Comment (davidtrevelyan) [reply compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp]: Reverted to traditional return type - thanks for the suggestion :)

**Pull request merged**